src/main/scala/me/rexim/issuestant/splittree/SplitIssueForest.scala

```diff
@@ -0,0 +1,37 @@
+package me.rexim.issuestant.splittree
+
+import me.rexim.issuestant.github._
+
+class SplitIssueForest(issues: List[Issue]) {
+  private def sortForest(forest: List[SplitIssueTree]): List[SplitIssueTree] =
+    forest.sortBy(_.number).map { tree =>
+      tree.copy(children = sortForest(tree.children))
+    }
+
+  private lazy val forest: List[SplitIssueTree] = sortForest {
+    val initialForestMap: Map[Int, SplitIssueTree] =
+      issues.map { issue =>
+        issue.number -> SplitIssueTree.fromIssue(issue)
+      }.toMap
+
+    val forestRootNumbers: Set[Int] =
+      issues
```

Comment:
Yes, you're totally right (I'm referring to 1). As I wrote this suggestion, I had the old non-implicit class `IssueWithParent` in mind.